### PR TITLE
Issue 25595 - multiple address virtual product error at checkout fixe…

### LINKED ISSUE #25595: Add all type of product into cart, start checkout with multiple address after giving multiple address, except download and virtual type product, remove all four type product from the cart then getting exception
### Preconditions (*)

1.Test environment:

* Magento version:2.3.3 or latest `2.4-develop`
* PHP Version:7.2
* mode: production
* Varnish:enable
* Js, CSS, HTML merged and minified
* Flat catalog/category – Yes
* Instance-  http://192.168.15.166/m2307/admin/
* username/password-admin/admin123

### Steps to reproduce (*)

1. add all six type of product into cart
2. proceed for checkout with multiple address
3. after give multiple address, remove four type product except download and virtual
4. getting exception

### Expected result (*)

1. it should not give exception

### Actual result (*)

1. Screenshot (7)

### PR DESCRIPTION
**Pull request by ralbin**
When you checkout with multiple addresses, and remove all the products that require shipping and you are only left with virtual products, no error is presented.  This sets a default message to it is more helpful.

### Description (*)

The class app/code/Magento/Multishipping/Controller/Checkout.php:127  is trying to get an error message that comes from core_config_data but no default message exists, causing the result to be null.
$error = $this->_getCheckout()->getMinimumAmountError();
This sets a default value allowing this render an appropriate message until one is created by the store owner.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#25595

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ralbin** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ralbin. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ralbin, here is your [new Magento instance](https://pr-29269.instances.magento-community.engineering).
Admin access: [https://pr-29269.instances.magento-community.engineering/admin_4ad7](https://pr-29269.instances.magento-community.engineering/admin_4ad7)
Login: `4d3fbbdc` Password: `7e42bc426163`
Instance will be terminated in up to 3 hours.

**Comment by ralbin** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ralbin. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ralbin, here is your [Magento instance](https://i-29269-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-29269-2-4-develop.instances.magento-community.engineering/admin_6e09](https://i-29269-2-4-develop.instances.magento-community.engineering/admin_6e09)
Login: `108e0663` Password: `3ab5558174e9`
Instance will be terminated in up to 3 hours.

**Comment by ralbin** [comment]
@magento run all tests

**Comment by ralbin** [comment]
@magento run Integration Tests

**Comment by ralbin** [comment]
@magento run Functional Tests EE

**Comment by ralbin** [comment]
@magento run all tests

**Comment by ralbin** [comment]
@magento run Integration Tests

**Comment by ralbin** [comment]
@magento run Integration Tests

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @ralbin. Thank you for collaboration. Can you please add a translation for the added message?

**Comment by ralbin** [comment]
@VladimirZaets I added an entry in app/code/Magento/Multishipping/i18n/en_US.csv:95  I hope that is what you expected.

**Comment by ralbin** [comment]
@magento run all tests

File: app/code/Magento/Multishipping/etc/config.xml [hunk 1]
```diff
@@ -13,5 +13,10 @@
                 <checkout_multiple_maximum_qty>100</checkout_multiple_maximum_qty>
             </options>
         </multishipping>
+        <sales>
+            <minimum_order>
+                <multi_address_error_message>All shippable products have been removed from your order</multi_address_error_message>
```
Comment (gabrieldagama) [inline app/code/Magento/Multishipping/etc/config.xml]: I think this message should be more general and just highlight an error, at the moment it sounds like the system has removed the items from the cart. I would recommend something like: "The current cart does not match multi shipping criteria, please review or contact the store administrator".

On the issue-specific flow, the check is being done is on `app/code/Magento/Multishipping/Helper/Data - isMultishippingCheckoutAvailable()` and that checks several different properties that are relevant for multi shipping.

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @ralbin, thanks for the pull request, it looks good! Please follow a small consideration on the error message, let me know your thoughts!

File: app/code/Magento/Multishipping/etc/config.xml (hunk 1 above)
Comment (ralbin) [reply app/code/Magento/Multishipping/etc/config.xml]: @gabrieldagama Adjusted and PR is ready for review.

**Comment by ralbin** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@ralbin thanks for the quick update! All our contributions have to be covered by tests, I think it would be great to cover it with MFTF test, let me know if you are available to cover it.

Thanks for your contribution!

**Comment by lbajsarowicz** [comment]
I think we can extend one of the existing tests in this case.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by gabrieldagama — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8025](https://jira.corp.magento.com/browse/ENGCOM-8025) has been created to process this Pull Request
:eight_spoked_asterisk: @gabrieldagama, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @ralbin . With PR changes, the message is displayed, but we cannot translate the phrase:
 `The current cart does not match multi shipping criteria, please review or contact the store administrator`.



Could you take a look?

Thanks!

**Comment by gabrieldagama** [comment]
Good catch @engcom-Alfa, @ralbin @engcom-Charlie I guess the problem here is that `app/code/Magento/Multishipping/Model/Checkout/Type/Multishipping - getMinimumAmountError` returns the raw message, instead of calling the translation function `__()` with the error msg.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by sidolov — APPROVED** [review]

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Create 1 Simple and 1 Virtual product;
2. Add both products to cart;
3. Proceed for checkout with multiple addresses;
4. Remove simple product except for virtual;

**Before:** :heavy_multiplication_x: An empty error message appears

Screenshot from 2020-08-19 09-31-05

**After:** :heavy_check_mark: The valid error message is shown

Screenshot from 2020-08-19 09-49-50

And we are able to translate this phrase:

**Pull request merged**